homeassistant/components/filesize/sensor.py

```diff
@@ -76,32 +75,22 @@ class Filesize(SensorEntity):
     def __init__(self, path: str) -> None:
         """Initialize the data object."""
         self._path = path  # Need to check its a valid path
-        self._size: int | None = None
-        self._last_updated: str | None = None
         self._attr_name = slugify(path.split("/")[-1]).strip().capitalize()
-        self._attr_unique_id = path
+        self._attr_extra_state_attributes = {}
 
     def update(self) -> None:
         """Update the sensor."""
-        statinfo = os.stat(self._path)
-        self._size = statinfo.st_size
-        self._last_updated = datetime.datetime.fromtimestamp(
-            statinfo.st_mtime
-        ).isoformat()
-
-    @property
-    def native_value(self) -> float | None:
-        """Return the size of the file in MB."""
-        decimals = 2
-        if self._size:
-            return round(self._size / 1e6, decimals)
-        return None
-
-    @property
-    def extra_state_attributes(self) -> dict[str, Any]:
-        """Return other details about the sensor state."""
-        return {
+        try:
+            statinfo = os.stat(self._path)
+        except OSError as error:
+            _LOGGER.error("Can not retrieve file statistics %s", error)
```

Comment:
You probably need to exit here. Maybe reset the native_value/state_attributes beforehand?
```suggestion
            _LOGGER.error("Can not retrieve file statistics %s", error)
            return
```